contrib/platform/src/com/sun/jna/platform/mac/IOKitUtil.java

```diff
@@ -0,0 +1,347 @@
+     *         conversion is lossy or the return value is out of range, then this
+     *         method returns an approximate value. If the value does not exist, the
+     *         {@code defaultValue} is returned.
+     */
+    public static int getIORegistryIntProperty(IORegistryEntry entry, String key, int defaultValue) {
+        int value = defaultValue;
+        CFStringRef keyAsCFString = CFStringRef.createCFString(key);
+        CFTypeRef valueAsCFType = IO.IORegistryEntryCreateCFProperty(entry, keyAsCFString, CF.CFAllocatorGetDefault(),
+                0);
+        if (valueAsCFType != null) {
+            CFNumberRef valueAsCFNumber = new CFNumberRef(valueAsCFType.getPointer());
+            value = valueAsCFNumber.intValue();
+        }
+        keyAsCFString.release();
+        if (valueAsCFType != null) {
+            valueAsCFType.release();
+        }
+        return value;
+    }
+
+    /**
+     * Convenience method to get a {@code double} value from an IO Registry.
+     *
+     * @param entry
+     *            A handle to the registry entry
+     * @param key
+     *            The string name of the key to retrieve
+     * @param defaultValue
+     *            The value to return if unsuccessful
+     * @return The value of the registry entry if it exists.
+     *         <p>
+     *         This method assumes a floating point value is stored and does not do
+     *         type checking. If this object's type differs from the return type,
+     *         and the conversion is lossy or the return value is out of range, then
+     *         this method returns an approximate value. If the value does not
+     *         exist, the {@code defaultValue} is returned.
+     */
+    public static double getIORegistryIntProperty(IORegistryEntry entry, String key, double defaultValue) {
+        double value = defaultValue;
+        CFStringRef keyAsCFString = CFStringRef.createCFString(key);
+        CFTypeRef valueAsCFType = IO.IORegistryEntryCreateCFProperty(entry, keyAsCFString, CF.CFAllocatorGetDefault(),
+                0);
+        if (valueAsCFType != null) {
+            CFNumberRef valueAsCFNumber = new CFNumberRef(valueAsCFType.getPointer());
+            value = valueAsCFNumber.doubleValue();
+        }
+        keyAsCFString.release();
+        if (valueAsCFType != null) {
+            valueAsCFType.release();
+        }
+        return value;
+    }
+
+    /**
+     * Convenience method to get a Boolean value from an IO Registry.
+     *
+     * @param entry
+     *            A handle to the registry entry
+     * @param key
+     *            The string name of the key to retrieve
+     * @param defaultValue
+     *            The value to return if unsuccessful
+     * @return The value of the registry entry if it exists; {@code defaultValue}
+     *         otherwise
+     */
+    public static boolean getIORegistryBooleanProperty(IORegistryEntry entry, String key, boolean defaultValue) {
```

Comment:
I noticed this while looking through the Unittests and just an idea: if the return and the default Value would be bound as the Object wrappers, people interested in the tristrate true/false/unset could pass in NULL as default and after the return do the check.

Reply:
Good thought. I just removed the defaults, and allowed plain null return to be consistent with the other types.  And found a copypaste typo because I didn't test the `double` one earlier.

Reply:
And I'm wondering if these might be better placed inside the `IORegistryEntry` class rather than a util.

Reply:
Given that the methods are tightly tied to the instance this sounds sane.

Reply:
OK... I'll do that tonight.  And add some ability to do some type checking as well, using the `CFTypeID` (`NativeLong`) of the registry entry, that maps to (static constant) `CFStringTypeID`, `CFNumberTypeID`, etc.